File: Iteration2/src/Domain/Registration.java
package Domain;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
@JsonIgnoreProperties(ignoreUnknown = true)
public class Registration implements Searchable{

    private String registrationNo;
    private String studentNo;
    private String advisorNo;
    private String courseCode;
    private RegistrationStatus status;
    private CourseSection courseSection;

    public Registration(String registrationNo, String studentNo, String advisorNo, String courseCode, RegistrationStatus status, CourseSection courseSection) {
        this.registrationNo = registrationNo;
        this.studentNo = studentNo;
        this.advisorNo = advisorNo;
        this.courseCode = courseCode;
        this.status = status;
        this.courseSection = courseSection;
    }

    public Registration(){}



    public String getRegistrationNo() {
        return registrationNo;
    }

    public void setRegistrationNo(String registrationNo) {
        this.registrationNo = registrationNo;
    }

    public String getStudentNo() {
        return studentNo;
    }

    public void setStudentNo(String studentNo) {
        this.studentNo = studentNo;
    }

    public String getAdvisorNo() {
        return advisorNo;
    }

    public void setAdvisorNo(String advisorNo) {
        this.advisorNo = advisorNo;
    }

    public String getCourseCode() {
        return courseCode;
    }

    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    public RegistrationStatus getStatus() {
        return status;
    }

    public void setStatus(RegistrationStatus status) {
        this.status = status;
    }

    public CourseSection getCourseSection() {
        return courseSection;
    }

    public void setCourseSection(CourseSection courseSection) {
        this.courseSection = courseSection;
    }
    

    @Override
    public String toString() {
        return "Registration{" +
                "registrationNo='" + registrationNo + '\'' +
                ", studentNo='" + studentNo + '\'' +
                ", advisorNo='" + advisorNo + '\'' +
                ", courseCode='" + courseCode + '\'' +
                ", status=" + status +
                '}';
    }
}

File: Iteration3/Domain/Registration.py
class Registration:
    def __init__(self, registration_no, student_no, advisor_no, course_code, status, course_section):
        self.registration_no = registration_no
        self.student_no = student_no
        self.advisor_no = advisor_no
        self.course_code = course_code
        self.status = status
        self.course_section = course_section

    # region Properties

    def get_registration_no(self):
        return self.registration_no

    def set_registration_no(self, registration_no):
        self.registration_no = registration_no

    def get_student_no(self):
        return self.student_no

    def set_student_no(self, student_no):
        self.student_no = student_no

    def get_advisor_no(self):
        return self.advisor_no

    def set_advisor_no(self, advisor_no):
        self.advisor_no = advisor_no

    def get_course_code(self):
        return self.course_code

    def set_course_code(self, course_code):
        self.course_code = course_code

    def get_status(self):
        return self.status

    def set_status(self, status):
        self.status = status

    def get_course_section(self):
        return self.course_section

    def set_course_section(self, course_section):
        self.course_section = course_section
    # endregion Properties

    def __str__(self):
        return f"Registration{{registration_no='{self.registration_no}', student_no='{self.student_no}', " \
               f"advisor_no='{self.advisor_no}', course_code='{self.course_code}', status={self.status}}}"



import json
from Domain.CourseSection import CourseSectionEncoder
from Domain.RegistrationStatus import RegistrationStatusEncoder

class RegistrationEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, Registration):
            return {
                'registration_no': obj.get_registration_no(),
                'student_no': obj.get_student_no(),
                'advisor_no': obj.get_advisor_no(),
                'course_code': obj.get_course_code(),
                'status': RegistrationStatusEncoder().default(obj.get_status()),
                'course_section': CourseSectionEncoder().default(obj.get_course_section()),
            }
        return super().default(obj)
